packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -255,27 +254,28 @@ function mountCustomElement(
     renderer: RendererAPI
 ) {
     const { sel, owner } = vnode;
-    const UpgradableConstructor = getUpgradableConstructor(sel, renderer);
+    const { createCustomElement } = renderer;
     /**
      * Note: if the upgradable constructor does not expect, or throw when we new it
      * with a callback as the first argument, we could implement a more advanced
      * mechanism that only passes that argument if the constructor is known to be
      * an upgradable custom element.
      */
     let vm: VM | undefined;
-    const elm = new UpgradableConstructor((elm: HTMLElement) => {
+
+    const upgradeCallback = (elm: HTMLElement) => {
         // the custom element from the registry is expecting an upgrade callback
         vm = createViewModelHook(elm, vnode, renderer);
-    });
+    };
+
+    const elm = createCustomElement(sel, upgradeCallback);
 
     linkNodeToShadow(elm, owner, renderer);
     vnode.elm = elm;
     vnode.vm = vm;
 
     if (vm) {
         allocateChildren(vnode, vm);
-    } else if (vnode.ctor !== UpgradableConstructor) {
-        throw new TypeError(`Incorrect Component Constructor`);
```

Comment:
There was no test for this error, and the tests pass when I remove it. Also, the refactor doesn't really allow for it. So it seemed safe to remove.

Reply:
well, not quite... I mean, the end goal is to remove it, but do we want to remove it now? I don't know.

Reply:
@caridy Are you still concerned about this?